modules/tinymce/src/core/main/ts/paste/Clipboard.ts

```diff
@@ -150,6 +150,24 @@ const getImagesFromDataTransfer = (editor: Editor, dataTransfer: DataTransfer):
   return Arr.filter(items.length > 0 ? items : files, isImage(editor));
 };
 
+const pasteImageFiles = (editor: Editor, files: File[], rng: Range | undefined): boolean => {
```

Comment:
I think more parts of `pasteImageData` needs to be moved into this function to do what the function name says. The `pasteImageData` will first grab the images from `getImagesFromDataTransfer` so it filters out images only. This function accepts any file type it should only take the images out so the `getImagesFromDataTransfer` should be split up into so that you can filter files here and in the `pasteImageData`. Also the boolean that this returns could be used to determine if the event should be prevented in `pasteImageData`. So some more refactoring is required here.

I would do it like this more or less:
* So I would do change the `getImagesFromDataTransfer` to `getFilesFromDataTransfer`
* Create a new function `getImagesFromFiles` that filters a collection of files based on isImage predicate
* Change `pasteImageFiles` to take files then run the `getImagesFromFiles`
* Then change the `pasteImageData` to prevent default if the `pasteImageFiles` returns true since then the clipboard has image data in it.

Reply:
refactored